enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+
+```bash
+$ oc adm credentials reconcile
+```
+
+This command would re-reconcile against the CredentialsRequests currently in the cluster. This is more of an on-going configuration management to ensure something doesn't change unexpectedly on us due to external factors. It would however support reducing permissions as a result of an additive only reconcile pre-upgrade.
+
+It should be noted that running this command after reconciling --available-updates, but before updating, would undo the pre-upgrade changes. As such this command should clear the reconciledAvailableUpdates list.
+
+
+## Open Questions [optional]
+
+ > 1. How can we handle net new CredentialsRequests in incoming release images?
+   * We could create the CredentialsRequest during the pre-upgrade reconciliation, but this does not guarantee that the target namespace for the component (where the Secret/ServiceAccount would be stored) exists.
+   * If we were to take the incoming CredentialsRequests, mind the credentials, and store in a temporary secret in the CCO namespace, we could later transfer it to the correct namespace when it exists.
+ > 1. Can the installer/CVO install manifests that contain status?
+   * If not we may need to hack around this by storing details on the minted credentials prior to installation as an annotation on the CredentialsRequest, and have the CCO reconcile it to it's final location in Status once we have a live object.
+   * Is it safe to edit a manifest from the release image prior to install?
+ > 1. Should the reconcile against current version command (i.e. no --available-updates flag) use the current CredentialsRequests in the cluster, or by extracting the current release image? The latter would allow us to prune net new CredentialsRequests processed for availableUpdates that were never installed.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Copying net new CredentialsRequests from future upgrades, minting credentials, storing them in a temporary location, and copying them to their final location during the upgrade is particularly nebulous.
+
+Involves a significant amount of new code.
+
+If included in "oc adm", would require vendoring and compiling basically all of cred operator so we can run the reconcile loop. (including all cloud provider libraries)
```

Comment:
@soltysh big question I had here, is this a decent fit for inclusion in oc adm, bearing in mind it means vendoring so much?

If not, is there some way we could distribute our own CLI tool to customers? Something in the CCO image perhaps.

Reply:
We're working on https://github.com/openshift/enhancements/pull/137 so that might be a reasonable option. I can't imagine vendoring CCO code in oc :sweat:

Reply:
Another reasonable option is to provide a new command to your operator and users could locally invoke `podman run cco mint-secrets --some-flags`. This ensures the code lives in only a single place, it's pretty straightforward and simple to document.

Reply:
> Another reasonable option is to provide a new command to your operator and users could locally invoke `podman run cco mint-secrets --some-flags`. This ensures the code lives in only a single place, it's pretty straightforward and simple to document.

I like that idea

Reply:
I like this as well.